src/testing/cmds/harness/test-collector/main.go

```diff
@@ -0,0 +1,182 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/log"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pctx"
+	"go.uber.org/zap"
+	"k8s.io/kubectl/pkg/util/slice"
+)
+
+type testOutput struct {
+	Time    time.Time
+	Action  string
+	Package string
+	Output  string
+}
+
+func main() {
+	log.InitPachctlLogger()
```

Comment:
Take a look at `InitBatchLogger`.  That will log to a file and only retain the file if there is an error.  Good for "I ran this command and cleared my screen", good news, there's a file ;)